### PR TITLE
Add paragraph on Scalameta

### PR DESCRIPTION
**Pull request by odersky**
I think we should add this section (or something like it) to the meta programming blog. @xeno-by @olafurpg do you have comments or suggestions to reword? I am not 100% sure of my characterization of differences between Scalameta and macros.

### CODE REVIEW

File: blog/_posts/2018-04-30-in-a-nutshell.md [hunk 1]
```diff
@@ -182,6 +182,13 @@ that so far required advanced macro code to define. In particular:
 
  - There will be a way to do typeclass derivation a la [Kittens](https://github.com/milessabin/kittens), [Magnolia](https://github.com/propensive/magnolia), or [scalaz-deriving](https://gitlab.com/fommil/scalaz-deriving) that does not need macros. We are currently evaluating the alternatives. The primary goal is to develop a scheme that is easy to use and that performs well at both compile- and run-time. A second goal is generality, as long as it does not conflict with the primary goal.
 
+## Meta Programming in the Large
```
Comment (olafurpg) [inline blog/_posts/2018-04-30-in-a-nutshell.md]: Thanks for adding this section! It is good to stress these are complementary technologies.

**Comment by SethTisue** [comment]
> I think we should add this section

👍, wish I'd thought to suggest it

File: blog/_posts/2018-04-30-in-a-nutshell.md [hunk 2]
```diff
@@ -182,6 +182,13 @@ that so far required advanced macro code to define. In particular:
...
+
+The future Scala 3 macro design is intended to replace the existing def macros and the `scala.reflect` infrastructure. But there is another meta programming system that is quite complementary to it: [Scalameta](http://scalameta.org/) provides high-quality syntactic analysis and code generation tools which are separate from the Scala compiler. As the name implies, Scalameta is run at the meta level, that is, it takes programs as input and produces syntactic or semantic information or rewritten programs as output. A macro system, by contrast, is integrated in the language and expands programs as they are compiled. There are potential synergies between the two projects. To name but two possibilities:
```
Comment (olafurpg) [inline blog/_posts/2018-04-30-in-a-nutshell.md]: > syntactic **and semantic** analysis

With SemanticDB it's possible to attribute Scalameta syntax trees with symbols and soon also types.

File: blog/_posts/2018-04-30-in-a-nutshell.md [hunk 3]
```diff
@@ -182,6 +182,13 @@ that so far required advanced macro code to define. In particular:
...
+
+ - Scalameta or projects derived from it such as [SemanticDB](https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md) could obtain type information directly from Tasty, which would make them independent from specific compilers.
```
Comment (olafurpg) [inline blog/_posts/2018-04-30-in-a-nutshell.md]: 👍

**Pull request merged**